### PR TITLE
KAFKA-4483; Fix NPE in `Log` constructor if log level is INFO or finer

### PR DESCRIPTION
**Pull request by ijuma**

### CODE REVIEW

**Comment by ijuma** [comment]
Working on adding a test.

**Comment by ijuma** [comment]
The reason the tests didn't catch it is that the logging level is ERROR. I think we should merge this change and then decide how to improve this.

**Comment by ijuma** [comment]
@junrao @hachikuji @ewencp @guozhangwang Any of you can review this? Without this fix, the broker is basically broken if logging level is INFO or finer.

File: core/src/main/scala/kafka/log/Log.scala [hunk 1]
```diff
@@ -104,19 +104,22 @@ class Log(@volatile var dir: File,
...
   locally {
     val startMs = time.milliseconds
+
     loadSegments()
+    /* Calculate the offset of the next message */
+    nextOffsetMetadata = new LogOffsetMetadata(activeSegment.nextOffset(), activeSegment.baseOffset,
+      activeSegment.size.toInt)
+
     info("Completed load of log %s with %d log segments and log end offset %d in %d ms"
       .format(name, segments.size(), logEndOffset, time.milliseconds - startMs))
```
Comment (ijuma) [inline core/src/main/scala/kafka/log/Log.scala:120]: The `logEndOffset` call here accesses `nextOffsetMetadata` which was `null` causing the NPE.

**Comment by vahidhashemian** [comment]
@ijuma Thanks for the quick fix. I no longer see the issue with this fix.

**Comment by hachikuji** [comment]
LGTM

**Pull request closed without merging**